tests/model/eventsources/test_self_managed_kafka_event_source.py

```diff
@@ -0,0 +1,165 @@
+                                "ec2:DescribeSecurityGroups",
+                            ],
+                            "Effect": "Allow",
+                            "Resource": "*",
+                        },
+                    ],
+                    "Version": "2012-10-17",
+                },
+                "PolicyName": "SelfManagedKafkaExecutionRolePolicy",
+            }
+        ]
+
+        self.assertEqual(policy_statements, expected_policy_document)
+
+    def test_must_raise_for_missing_topics(self):
+        self.kafka_event_source.SourceAccessConfigurations = [
+            {"Type": "SASL_SCRAM_256_AUTH", "URI": "SECRET_URI"},
+            {"Type": "VPC_SUBNET", "URI": "SECRET_URI"},
+            {"Type": "VPC_SECURITY_GROUP", "URI": "SECRET_URI"},
+        ]
+        self.kafka_event_source.KafkaBootstrapServers = ["endpoint1", "endpoint2"]
+        self.kafka_event_source.Enabled = True
+        self.kafka_event_source.BatchSize = 1
+
+        with self.assertRaises(InvalidEventException):
+            self.kafka_event_source.get_policy_statements()
+
+    def test_must_raise_for_multiple_topics(self):
+        self.kafka_event_source.SourceAccessConfigurations = [
+            {"Type": "SASL_SCRAM_256_AUTH", "URI": "SECRET_URI"},
+            {"Type": "VPC_SUBNET", "URI": "SECRET_URI"},
+            {"Type": "VPC_SECURITY_GROUP", "URI": "SECRET_URI"},
+        ]
+        self.kafka_event_source.KafkaBootstrapServers = ["endpoint1", "endpoint2"]
+        self.kafka_event_source.Topics = ["Topics1", "Topics2"]
+        self.kafka_event_source.Enabled = True
+        self.kafka_event_source.BatchSize = 1
+
+        with self.assertRaises(InvalidEventException):
+            self.kafka_event_source.get_policy_statements()
+
+    def test_must_raise_for_missing_endpoints(self):
+        self.kafka_event_source.SourceAccessConfigurations = [
+            {"Type": "SASL_SCRAM_256_AUTH", "URI": "SECRET_URI"},
+            {"Type": "VPC_SUBNET", "URI": "SECRET_URI"},
+            {"Type": "VPC_SECURITY_GROUP", "URI": "SECRET_URI"},
+        ]
+        self.kafka_event_source.Enabled = True
+        self.kafka_event_source.Topics = ["Topics"]
+        self.kafka_event_source.BatchSize = 1
+
+        with self.assertRaises(InvalidEventException):
+            self.kafka_event_source.get_policy_statements()
+
+    def test_must_raise_for_missing_vpc_config(self):
+        self.kafka_event_source.SourceAccessConfigurations = [
+            {"Type": "SASL_SCRAM_256_AUTH", "URI": "SECRET_URI"},
+            {"Type": "VPC_SUBNET", "URI": "SECRET_URI"},
+        ]
+        self.kafka_event_source.KafkaBootstrapServers = ["endpoint1", "endpoint2"]
+        self.kafka_event_source.Enabled = True
+        self.kafka_event_source.Topics = ["Topics"]
+        self.kafka_event_source.BatchSize = 1
+
+        with self.assertRaises(InvalidEventException):
+            self.kafka_event_source.get_policy_statements()
```

Comment:
Could you add the following tests:
- `SourceAccessConfigurations`: fail for None or [] or missing `VPC_SUBNET`
- `KafkaBootstrapServers`: fail for []
- `Topics`: fail for []
- kms policy
- Unknown `SourceAccessConfigurations` `Type`?
- Invalid URIs for each SourceAccessConfigurations Type